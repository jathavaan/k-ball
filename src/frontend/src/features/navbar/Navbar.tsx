import {
  StyledAppBar,
  StyledToolbar,
  LogoButton,
  LogoImage,
} from "./navbar.style";
import logo from "../../assets/Logo.png";
import { useNavigate } from "react-router-dom";

export function Navbar() {
  const navigate = useNavigate();
  return (
    <StyledAppBar position="static">
      <StyledToolbar>
        <LogoButton onClick={() => navigate("/project2")}>
          <LogoImage src={logo} alt="Logo" />
        </LogoButton>
      </StyledToolbar>
    </StyledAppBar>
  );
}
