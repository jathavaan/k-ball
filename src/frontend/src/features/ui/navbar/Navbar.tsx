import "./navbar.css";

export const Navbar = () => {
  return <h1>Project 2</h1>;
};
