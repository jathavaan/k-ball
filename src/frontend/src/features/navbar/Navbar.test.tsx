import { render, screen } from "@testing-library/react";
import { Navbar } from "./Navbar";
import logo from "../../assets/Logo.png";

describe("Navbar", () => {
  it("should render the logo image", () => {
    render(<Navbar />);
    expect(screen.getByAltText("Logo")).toBeDefined();
  });

  it("Should render the logo image with correct src", () => {
    render(<Navbar />);
    const logoImage = screen.getByAltText("Logo") as HTMLImageElement;
    expect(logoImage.src).toContain(logo);
  });

  it("should match Navbar snapshot", () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
