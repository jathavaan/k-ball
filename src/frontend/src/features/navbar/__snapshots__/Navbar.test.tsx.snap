// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should match Navbar snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1uphetq-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1bfkwbn-MuiToolbar-root"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-fdexxj-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
         
        <img
          alt="Logo"
          class="css-88a8ul"
          src="/project2/src/assets/Logo.png"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
